docs/source/examples/Widget Custom.ipynb

```diff
@@ -0,0 +1,785 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "The TypeScript cookiecutter generates a file `src/widget.ts`. Open the file and rename `ExampleModel` to `EmailModel` and `ExampleView` to `EmailView`:\n",
+    "\n",
+    "```typescript\n",
+    "export class EmailModel extends DOMWidgetModel {\n",
+    "  defaults() {\n",
+    "    return {...super.defaults(),\n",
+    "      _model_name: EmailModel.model_name,\n",
+    "      _model_module: EmailModel.model_module,\n",
+    "      _model_module_version: EmailModel.model_module_version,\n",
+    "      _view_name: EmailModel.view_name,\n",
+    "      _view_module: EmailModel.view_module,\n",
+    "      _view_module_version: EmailModel.view_module_version,\n",
+    "      value : 'Hello World'\n",
+    "    };\n",
+    "  }\n",
+    "\n",
+    "  static serializers: ISerializers = {\n",
+    "      ...DOMWidgetModel.serializers,\n",
+    "      // Add any extra serializers here\n",
+    "    }\n",
+    "\n",
+    "  static model_name = 'EmailModel';\n",
+    "  static model_module = MODULE_NAME;\n",
+    "  static model_module_version = MODULE_VERSION;\n",
+    "  static view_name = 'EmailView';\n",
+    "  static view_module = MODULE_NAME;\n",
+    "  static view_module_version = MODULE_VERSION;\n",
+    "}\n",
+    "\n",
+    "\n",
+    "export class EmailView extends DOMWidgetView {\n",
+    "  render() {\n",
+    "    this.el.classList.add('custom-widget');\n",
+    "\n",
+    "    this.value_changed();\n",
+    "    this.model.on('change:value', this.value_changed, this);\n",
+    "  }\n",
+    "\n",
+    "  value_changed() {\n",
+    "    this.el.textContent = this.model.get('value');\n",
+    "  }\n",
+    "}\n",
+    "\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "### Render method"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, override the base `render` method of the view to define custom rendering logic.\n",
+    "\n",
+    "A handle to the widget's default DOM element can be acquired via `this.el`.  The `el` property is the DOM element associated with the view.\n",
+    "\n",
+    "In `src/widget.ts`, define the `_emailInput` attribute:\n",
+    "\n",
+    "```typescript\n",
+    "export class EmailView extends DOMWidgetView {\n",
+    "  private _emailInput: HTMLInputElement;\n",
+    "}\n",
+    "```\n",
```

Comment:
I found this confusing. It made it seem that that there shouldn't be anything in the brackets, i.e. delete the render method.

Perhaps better would be
```
export class EmailView extends DOMWidgetView {
  private _emailInput: HTMLInputElement;

  render() {
     .....
}

Reply:
Good point, adding that :+1: